File: cloud/workflow-server/src/main/java/pers/acp/admin/workflow/constant/WorkFlowExpression.java
package pers.acp.admin.workflow.constant;

import pers.acp.admin.common.constant.ModuleFuncCode;
import pers.acp.admin.common.constant.RoleCode;
import pers.acp.admin.common.permission.BaseExpression;

/**
 * @author zhang by 14/06/2019
 * @since JDK 11
 */
public interface WorkFlowExpression extends BaseExpression {

    /**
     * 启动流程
     */
    String flowStart = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.flowStart + "')";

    /**
     * 获取流程待办任务
     */
    String flowPending = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.flowPending + "')";

    /**
     * 审批
     */
    String flowApprove = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.flowApprove + "')";

    /**
     * 获取流程处理历史
     */
    String flowHistory = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.flowHistory + "')";

    /**
     * 获取流程图
     */
    String flowDiagram = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.flowDiagram + "')";

}

File: cloud/workflow-server/src/main/kotlin/pers/acp/admin/workflow/constant/WorkFlowExpression.kt
package pers.acp.admin.workflow.constant

import pers.acp.admin.constant.ModuleFuncCode
import pers.acp.admin.constant.RoleCode

/**
 * @author zhang by 14/06/2019
 * @since JDK 11
 */
object WorkFlowExpression {
    /**
     * 流程定义
     */
    const val flowDefinition =
        "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.flowDefinition + "')"

    /**
     * 流程管理
     */
    const val flowAdmin =
        "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.flowAdmin + "')"

    /**
     * 获取流程待办任务
     */
    const val flowPending =
        "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.flowPending + "')"

    /**
     * 流程处理
     */
    const val flowProcess =
        "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.flowAdmin + "','" + ModuleFuncCode.flowProcess + "')"
}